server/channels/store/sqlstore/post_store.go

```diff
@@ -164,6 +165,29 @@ func (s *SqlPostStore) SaveMultiple(rctx request.CTX, posts []*model.Post) ([]*m
 		}
 		post.ValidateProps(rctx.Logger())
 
+		if post.Type == model.PostTypeBurnOnRead {
+			expireAt := post.GetProp(model.PostPropsExpireAt)
+			if expireAt == "" {
+				return nil, idx, errors.New("expire_at is required for burn on read posts")
+			}
+
+			if post.RootId != "" {
```

Comment:
Not sure if this is already handled in another PR, but should we also error out when trying to create a reply to a BoR post?